packages/plugins/embarkjs/src/index.js

```diff
@@ -46,11 +57,62 @@ class EmbarkJS {
     this.events.setCommandHandler("embarkjs:contract:generate", this.addContractArtifact.bind(this));
     this.events.setCommandHandler("embarkjs:contract:runInVm", this.runInVm.bind(this));
 
+    if (!this.enabled) return;
+
     embark.registerActionForEvent("pipeline:generateAll:before", this.addEmbarkJSArtifact.bind(this));
     embark.registerActionForEvent("pipeline:generateAll:before", this.addContractIndexArtifact.bind(this));
+
+    if (this.config.blockchainConfig.enabled) {
+      this.setupEmbarkJS();
+    }
+
+    embark.registerActionForEvent("deployment:contract:deployed", {priority: 40}, this.registerInVm.bind(this));
+    embark.registerActionForEvent("deployment:contract:undeployed", this.registerInVm.bind(this));
+    embark.registerActionForEvent("deployment:contract:deployed", this.registerArtifact.bind(this));
+    embark.registerActionForEvent("deployment:contract:undeployed", this.registerArtifact.bind(this));
+  }
+
+
+
+  async setupEmbarkJS() {
+    if (!this.enabled) return;
+    this.events.on("blockchain:started", async () => {
+      await this.registerWeb3Object();
+      this.events.request("embarkjs:console:setProvider", 'blockchain', 'web3', '{web3}');
+    });
+    this.events.request("embarkjs:plugin:register", 'blockchain', 'web3', 'embarkjs-web3');
+    await this.events.request2("embarkjs:console:register", 'blockchain', 'web3', 'embarkjs-web3');
+  }
+
+  async registerWeb3Object() {
+    if (!this.enabled) return;
+    const provider = await this.events.request2("blockchain:client:provider", "ethereum");
+    const web3 = new Web3(provider);
+    this.events.request("runcode:whitelist", 'web3', () => {});
+    await this.events.request2("runcode:register", 'web3', web3);
+    const accounts = await web3.eth.getAccounts();
+    if (accounts.length) {
+      await this.events.request2('runcode:eval', `web3.eth.defaultAccount = '${accounts[0]}'`);
+    }
+
+    this.events.request('console:register:helpCmd', {
+      cmdName: "web3",
+      cmdHelp: __("instantiated web3.js object configured to the current environment")
+    }, () => {});
+  }
+
+  async registerInVm(params, cb) {
+    if (!this.enabled) return;
+    this.events.request("embarkjs:contract:runInVm", params.contract, cb);
+  }
+
+  registerArtifact(params, cb) {
+    if (!this.enabled) return;
```

Comment:
Ok, you use that condition a lot. Might be worth checking if the function is actually called in a context where the condition is needed or if it's always called when it is enabled.

Reply:
it's there just to avoid calling it, while keeping the same setCommandHandlers and others for backwards compatibility. this plugin is intended to disapear in the short term.